Source/WebGPU/WebGPU/Queue.mm

```diff
@@ -34,12 +34,35 @@
 #import "Texture.h"
 #import <wtf/CheckedArithmetic.h>
 
+@interface TextureSliceLevel : NSObject
+- (instancetype)initWithTexture:(id<MTLTexture>)texture slice:(NSUInteger)slice level:(NSUInteger)level;
+@property (nonatomic) id<MTLTexture> texture;
+@property (nonatomic) NSUInteger slice;
+@property (nonatomic) NSUInteger level;
+@end
+
+@implementation TextureSliceLevel
+- (instancetype)initWithTexture:(id<MTLTexture>)texture slice:(NSUInteger)slice level:(NSUInteger)level
+{
+    if (!(self = [super init]))
+        return nil;
+
+    _texture = texture;
+    _slice = slice;
+    _level = level;
+
+    return self;
+}
+@end
+
 namespace WebGPU {
 
 Queue::Queue(id<MTLCommandQueue> commandQueue, Device& device)
     : m_commandQueue(commandQueue)
     , m_device(device)
 {
+    m_modifiedResources = [NSMutableSet set];
+    m_modifiedTextures = [NSMutableSet set];
```

Comment:
Can we use `Vector<RetainPtr<MTLResource>>` instead? We generally prefer WTF types in WebKit.

Reply:
I can use the WTF type but then to use HashSet I think id<MTLTexture> needs a hash function

Reply:
I didn't want to use Vector to avoid searching for duplicates or encoding multiple synchronization calls

but I'm checking to see if these calls are even needed after fixing the other race condition